### PR TITLE
Add more tests for `publish_entry` and `get_next_entry_args`

### PR DESCRIPTION
**Pull request by sandreae**
- [x] test for update of a different authors document (multi-writer update)
- [x] test for passing non-existent document id

This just adds some more tests for an area where there seems to be quite a few edge cases/tricky behaviour. We don't think `get_entry_args` is performing as we want when being passed non-existent `DocumentId`s atm. 

## 📋 Checklist

- [x] Add tests that cover your changes
- [ ] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa)

### CODE REVIEW

**Comment by adzialocha** [comment]
I think we should add a changelog entry for anything which can't be expressed as a single commit directly into `main` 😅

File: p2panda-rs/src/storage_provider/traits/storage_provider.rs [hunk 1]
```diff
@@ -541,6 +536,114 @@ pub mod tests {
...
+        let expected_entry_args_response =
+            EntryArgsResponse::new(None, None, SeqNum::new(1).unwrap(), LogId::new(1));
+
+        // Correct request returns the expected next entry args.
+        assert_eq!(entry_args_response, expected_entry_args_response);
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn get_entry_args_non_existand_document_id(
```
Comment (adzialocha) [inline p2panda-rs/src/storage_provider/traits/storage_provider.rs:620]: "existand" typo

File: p2panda-rs/src/storage_provider/traits/storage_provider.rs [hunk 2]
```diff
@@ -541,6 +536,114 @@ pub mod tests {
...
+        let entry_args_request = EntryArgsRequest {
+            public_key: Author::try_from(key_pair.public_key().to_owned()).unwrap(),
+            document_id: Some(document_id),
+        };
+
+        let entry_args_response = db.store.get_entry_args(&entry_args_request).await.unwrap();
+
+        // The response contains args for a new log at sequence number 1.
+        //
+        // @TODO: This is the current behaviour but it may not be what we want.
```
Comment (adzialocha) [inline p2panda-rs/src/storage_provider/traits/storage_provider.rs:640]: Could be nice to clarify what is the expected behavior. Is there a ticket?

**Review by adzialocha — CHANGES_REQUESTED** [review]
Small things plus the changelog comment above ✌️
